openhcl/openhcl_boot/src/hypercall.rs

```diff
@@ -85,47 +67,60 @@ impl HvCall {
     /// necessary.
     #[cfg(target_arch = "x86_64")]
     pub fn hypercall_page(&mut self) -> u64 {
-        self.init_if_needed();
+        assert!(self.initialized);
         core::ptr::addr_of!(minimal_rt::arch::hypercall::HYPERCALL_PAGE) as u64
     }
 
-    fn init_if_needed(&mut self) {
-        if !self.initialized {
-            self.initialize();
-        }
-    }
-
-    pub fn initialize(&mut self) {
-        assert!(!self.initialized);
+    /// Hypercall initialization.
+    pub fn initialize(&mut self, isolation_type: IsolationType) {
+        match isolation_type {
+            #[cfg(target_arch = "x86_64")]
+            IsolationType::Tdx => {
+                self.initialize_tdx();
+            }
+            #[cfg(target_arch = "x86_64")]
+            IsolationType::Snp => (),
+            _ => {
+                if !self.initialized {
```

Comment:
this `initiialized` check seems like it should be hoisted out of `initialize_tdx` and into this outer function?

Reply:
`initialize_tdx` can still assert it's not initialized